artifactory/system.go

```diff
@@ -33,6 +40,1112 @@ func (v Versions) String() string {
+	AllowBlockedArtifactsDownload *bool   `yaml:"allowBlockedArtifactsDownload,omitempty" xml:"allowBlockedArtifactsDownload,omitempty"`
+	BypassDefaultProxy            *bool   `yaml:"bypassDefaultProxy,omitempty" xml:"bypassDefaultProxy,omitempty"`
+	Proxy                         *string `yaml:"proxy,omitempty" xml:"proxy,omitempty"`
+}
+
+// BintrayConfig represents Bintray settings in Artifactory's Configuration
+// Descriptor. This is undocumented in YAML Configuration File.
+//
+type BintrayConfig struct {
+	UserName        *string `yaml:"userName,omitempty" xml:"userName,omitempty"`
+	ApiKey          *string `yaml:"apiKey,omitempty" xml:"apiKey,omitempty"`
+	FileUploadLimit *int    `yaml:"fileUploadLimit,omitempty" xml:"fileUploadLimit,omitempty"`
+}
+
+// Proxy represents a Proxy setting in Artifactory's Global Configuration Descriptor.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-General(General,PropertySets,Proxy,Mail)
+type Proxy struct {
+	Key             *string `yaml:"-" xml:"key,omitempty"`
+	Domain          *string `yaml:"domain,omitempty" xml:"domain,omitempty"`
+	Host            *string `yaml:"host,omitempty" xml:"host,omitempty"`
+	NtHost          *string `yaml:"ntHost,omitempty" xml:"ntHost,omitempty"`
+	Password        *string `yaml:"password,omitempty" xml:"password,omitempty"`
+	Port            *int    `yaml:"port,omitempty" xml:"port,omitempty"`
+	RedirectToHosts *string `yaml:"redirectedToHosts,omitempty" xml:"redirectedToHosts,omitempty"`
+	Username        *string `yaml:"username,omitempty" xml:"username,omitempty"`
+	DefaultProxy    *bool   `yaml:"defaultProxy,omitempty" xml:"defaultProxy,omitempty"`
+}
+
+// Proxies is an alias for a slice of Proxy that can be
+// properly marshaled to/from YAML.
+type Proxies []*Proxy
+
+// MarshalYAML implements the yaml.Marshaller interface for Proxies.
+func (p Proxies) MarshalYAML() (interface{}, error) {
+	proxiesMap := make(map[string]*Proxy)
+	for _, proxy := range p {
+		if *proxy == (Proxy{Key: proxy.Key}) {
+			proxiesMap[*proxy.Key] = nil
+		} else {
+			proxiesMap[*proxy.Key] = proxy
+		}
+
+	}
+
+	return proxiesMap, nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Proxies.
+func (p *Proxies) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	proxiesMap := make(map[string]*Proxy)
+	if err := unmarshal(proxiesMap); err != nil {
+		return err
+	}
+
+	var proxiesSlice Proxies
+	for proxyKey, proxy := range proxiesMap {
+		if proxy == nil {
+			continue
+		}
+		proxy.Key = &proxyKey
+		proxiesSlice = append(proxiesSlice, proxy)
+	}
+
+	*p = proxiesSlice
+	return nil
+}
+
+// ReverseProxy represents a Reverse Proxy configuration in Artifactory's Global Configuration Descriptor.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-GetReverseProxyConfiguration
+type ReverseProxy struct {
+	Key                      *string `yaml:"-" xml:"key,omitempty"`
```

Comment:
I noticed we're not allowing this field to be serialized in YAML due to the `-` tag.

This peaked my curiosity so was this intentional?

Reply:
Similar to [comment](https://github.com/target/go-arty/pull/57#discussion_r436119180).